File: ios/ReactNativeEventEmitter.m
//
//  ReactNativeEventEmitter.m
//  IrisLibrary
//
//  Created by Thiago Brandt on 05/05/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(ReactNativeEventEmitter, RCTEventEmitter)
RCT_EXTERN_METHOD(supportedEvents)
@end

File: ios/ReactNativeEventEmitter.swift
//
//  ReactNativeEventEmitter.swift
//  IrisLibrary
//
//  Created by Thiago Brandt on 18/05/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation

open class ReactNativeEventEmitter: RCTEventEmitter {
    
    override init(){
        super.init()
        IrisEventEmitter.instance.register(self)
    }
    @objc
    open override func supportedEvents() -> [String] {
        return IrisEventEmitter.instance.allEvents
    }
}
